nixpkgs_review/report.py

```diff
@@ -39,6 +40,21 @@ def html_pkgs_section(packages: List[Attr], msg: str, what: str = "package") ->
     return res
 
 
+def md_pkgs_generic_test(packages: List[Attr]) -> str:
+    if len(packages) == 0:
+        return ""
+    res = "\n"
+    for pkg in packages:
+        if subprocess.call([pkg.path, "--help"],
```

Comment:
This sounds like good idea, however:

- It's actually a bit scary to run arbitrary programs. I know this was done by https://github.com/ryantm/nixpkgs-update however this is a dedicated account not having access to the user's home. 
- Some programs might also hang indefinitely...

We could rely on the build sandbox if available to limit the scope but also in this case a timeout is needed (I think nix-build supports that).

Reply:
- [ ] I just added the timeout
- [ ] Could you guide me on how to ensure this is only done in sandboxes?
- [ ] Please add a commit to this fixing the mocks for the tests to go green, I've tried it for nearly 5 hours with no success. :pray:

Reply:
I think the cleanest way would be to do it as part of the normal build process. During the build we write a `build.nix`. This needs to be extended to have derivation like:

```nix
{ pkgs }:
let
  checkCommands = pkgs.callPackage "/absolute/path/to/nixpkgs_review/nixpkgs_review/nix/checkCommands.nix" {};
in {
  helloCheck = runCommand "check-hello" {} ''
    ${checkCommands}/bin/checkCommands ${hello} $out/report.json
  '';
}
```

`checkCommands` could be a python script that does the subprocess stuff that you are already doing and than write a json file. You can use `nix eval` to get derivation paths out of `build.nix` and than read the json from there. Does this make sense? It is not as straight forward as the current implementation however I don't want to execute arbitrary code on the reviewers private machine.

Reply:
If you write the code I can take care of the tests afterwards.

Reply:
Another thing that is interesting is to catch segfaults. subprocess should have some exception in case signals such as SIGSEV are thrown.

Reply:
The idea with SIGSEV isn't exposed by `timout`. I didn't want to use python, since it was already sufficently omplex to get my head arround the way this derivation is expected to be written. I figured with bash it would be easier for a start.